caffeine/core.py

```diff
@@ -120,16 +127,52 @@ def __attempt_autoactivation(self):
                 elif not self.get_activated():
                     logger.info("Fullscreen app detected. Inhibiting.")
 
-        if (process_running or fullscreen) and not self.__auto_activated:
+        # Let's look for playing audio:
+        self.music_procs = 0  # Number of supposed audio only streams. We can turn the screen off for those
+        screen_relevant_procs = 0  # Number of all audio streams including videos. We want to keep the screen on here
+
+        if not process_running and not fullscreen:
+            # Get all audio playback streams
+            audio_sinks = os.popen("pacmd list-sink-inputs").read()
```

Comment:
I'm concerned about adding `pacmd` as an extra dependency.
Is there some way of reading this without it (eg: some PA socket? or a PA python library?).

Reply:
> I'm concerned about adding `pacmd` as an extra dependency.
> Is there some way of reading this without it (eg: some PA socket? or a PA python library?).

At least on Arch Linux pacmd is part of pulseaudio itself. You cannot use pulseaudio without having this tool onboard too.

Reply:
I don't use ALSA directly, but users who do will be locked out of this feature. Is there a way to do this with only ALSA? If so, all users will benefit. 

Afaik pulseaudio sits on top on ALSA, since ALSA can only handle one audio stream at a time and needs another tool to do the mixup of multiple streams into one. As far as I can tell almost no one uses ALSA without pulseaudio. And also: This pull request is intended to solve the issues for my use case (which should be congruent with the vast majority of users). ALSA only users don't gain anything here, that's right. But they also don't lose any functionality. So I don't intend to invest time to make this possible (anyone else is invited to do so though).

The thing with the pulsectl library is very much doable though. Thanks for the input

Reply:
> If you want to avoid dbus, you could use a library such as python-pulse-control. 

I was having a look on the library. While it works flawlessy when retrieving the audio sinks and sources, it doesn't seem to have an (easy) access the the `state: RUNNING` property of a source/sink, which seems to be provided by the `PA_SINK_INPUT_INFO` class in the pulsectl library. I need this in order to determine to deactivate suspending. The reason:

When you pause the audio playback, the audio sink/source (depending on the application) still remains in the list, but the state switches to `state: CORKED`. So in order to allow power suspension while paused audio playback applications are running, you need to have a look on this attribute.

So if no one can point me how to get this attribute via pulsectl, I would stick the solution which is implemented right now.

Reply:
This looks okay to me -- I'm guessing `pactl` will have the same issues regarding Chrome, right? Since it probably just uses the same dbus interface.

Yeah, firefox has the same issue actually (and it keeps my headphones from powering down when inactive :cry: ).

I'd be happy to merge the PR using this interface -- it also looks much cleaner.

Reply:
Also, this expects users to be using pulseaudio, if it turns out many people have an issue with this, I'll look into making it optional.

But for now, I'd avoid that effor, since it may not be an actual usecase anyway.

Reply:
> Luckily for you, I spent hours figuring it all out with basically no documentation (I should help get that fixed if the dev wants more people to use it)

Haha, I thought the same when having a look. But thank you sitting through the pain. I've implemented it now and updated the pull request